### PR TITLE
Use new JSX transform

### PR DESCRIPTION
**Pull request by tbergq**
https://reactjs.org/blog/2020/09/22/introducing-the-new-jsx-transform.html

### CODE REVIEW

File: src/eslint-config-adeira/ourRules.js [hunk 1]
```diff
@@ -429,7 +429,7 @@ module.exports = ({
   'react/prefer-read-only-props': OFF,
   'react/prefer-stateless-function': OFF,
   'react/prop-types': OFF, // we use Flow instead,
-  'react/react-in-jsx-scope': WARN,
+  'react/react-in-jsx-scope': OFF,
```
Comment (tbergq) [inline src/eslint-config-adeira/ourRules.js:432]: Possibly we should leave this and just update the rule directly in `.eslintrc.js`, and change it here with the next major version

Comment (mrtnzlml) [reply src/eslint-config-adeira/ourRules.js:432]: File `src/eslint-config-adeira/getCommonConfig.js` has `detectReactVersion` config. Would that be helpful?

Comment (tbergq) [reply src/eslint-config-adeira/ourRules.js:432]: But using the new `JSX` transform is still optional, so I don't think so.

File: babel.config.js [hunk 2]
```diff
@@ -35,6 +35,7 @@ module.exports = function (api /*: ApiType */) /*: BabelConfig */ {
         '@adeira/babel-preset-adeira',
         {
           target: api.caller(isWebpack) ? 'js-esm' : 'js',
+          reactRuntime: 'automatic',
```
Comment (mrtnzlml) [inline babel.config.js:38]: We should probably swich it to `automatic` by default after this PR. Followup issue? :)

Comment (tbergq) [reply babel.config.js:38]: #1253

File: src/relay/CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,7 @@
 # Unreleased
 
+- Change to use the new [JSX transforms](https://reactjs.org/blog/2020/09/22/introducing-the-new-jsx-transform.html)
```
Comment (mrtnzlml) [inline src/relay/CHANGELOG.md:3]: Are there any implications for the user-land code? Does it require to use the new JSX transform as well?

Comment (tbergq) [reply src/relay/CHANGELOG.md:3]: No, it does not. But it needs to be using a version compatible with the new `JSX` transform. So 16.14.0 or greater

Comment (tbergq) [reply src/relay/CHANGELOG.md:3]: The transpilde code looks like 

```jsx
import {jsx as _jsx} from 'react/jsx-runtime';

function App() {
  return _jsx('h1', { children: 'Hello world' });
}
```

So as long as that import path exists in react, the consumer is fine.

File: src/sx-tailwind-website/__github__/babel.config.js [hunk 4]
```diff
@@ -10,8 +10,8 @@ type ApiType = {|
 |};
 
 type BabelConfig = {|
-  +presets: $ReadOnlyArray<string>,
-  +plugins: $ReadOnlyArray<string>,
+  +presets: $ReadOnlyArray<string | [string, { ... }]>,
+  +plugins: $ReadOnlyArray<string >,
```
Comment (mrtnzlml) [inline src/sx-tailwind-website/__github__/babel.config.js:14]:
```suggestion
  +plugins: $ReadOnlyArray<string>,
```

File: src/eslint-config-adeira/ourRules.js (hunk 1 above)
Comment (tbergq) [reply src/eslint-config-adeira/ourRules.js:432]: We are just flipping it from `WARN` -> `OFF`, so I think this approach is fine.

File: src/sx-tailwind-website/__github__/babel.config.js (hunk 4 above)
Comment (tbergq) [reply src/sx-tailwind-website/__github__/babel.config.js]: Done

**Review by mrtnzlml — APPROVED** [review]

**Pull request merged**